### PR TITLE
Create a DApp Tutorial

### PR DESCRIPTION
**Pull request by jboetticher**
### Description

A beginner tutorial that helps someone learn the components of every DApp. Might need a change in the section with private keys

### Checklist

- [ ] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [X] If images have been added, I have run the `compress-images.py` script to compress the images.
- [ ] If this page requires a disclaimer, I have added one

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

### CODE REVIEW

File: tutorials/eth-api/complete-dapp-architecture.md [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+    {/* Wrapper Components */}
+  )
+}
+```
+
+And, if you've done any transactions, you'll see that they'll pop up!  
+
+![Finished DApp](/images/tutorials/eth-api/complete-dapp/complete-dapp-5.png)
+
+Now you've implemented 3 main components of DApp frontends: reading from storage, sending transactions, and reading logs. With these building blocks as well as the knowledge you learned with smart contracts and nodes you should be able to cover 80% of DApps. Of course, there are more advanced (but optional) components of DApps that have popped up over time, some of which are listed in the next section.
```
Comment (eshaben) [inline tutorials/eth-api/complete-dapp-architecture.md:693]:
```suggestion
Now you've implemented three main components of DApp frontends: reading from storage, sending transactions, and reading logs. With these building blocks as well as the knowledge you learned with smart contracts and nodes you should be able to cover 80% of DApps. Of course, there are more advanced (but optional) components of DApps that have popped up over time, some of which are listed in the next section.  
```

File: tutorials/eth-api/complete-dapp-architecture.md [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+}
+```
+
+And, if you've done any transactions, you'll see that they'll pop up!  
+
+![Finished DApp](/images/tutorials/eth-api/complete-dapp/complete-dapp-5.png)
+
+Now you've implemented 3 main components of DApp frontends: reading from storage, sending transactions, and reading logs. With these building blocks as well as the knowledge you learned with smart contracts and nodes you should be able to cover 80% of DApps. Of course, there are more advanced (but optional) components of DApps that have popped up over time, some of which are listed in the next section.  
+
+## Additional Options {: #additional-options }
```
Comment (eshaben) [inline tutorials/eth-api/complete-dapp-architecture.md:695]: This section (and subsections) seems out of place to me. Also some of this is covered in the Web2 vs Web3 blogpost that should have been published months ago haha.. I'm thinking maybe we can remove it from here and just link to the blogpost so we keep this as more of a code-oriented guide 🤷‍♀️ 

I'll try to figure out when the blogpost will actually be published

**Review by eshaben — CHANGES_REQUESTED** [review]
Another thorough tutorial! Nice job!

Some thoughts:
- The index page icon is missing for the tutorial
- I think for SEO purposes, it would be beneficial to rename the tutorial "How to Build a DApp" - what are your thoughts?
- I didn't do the full grammar check yet but I can do that after some initial changes are made
- This guide could benefit from diagrams of the architecture (made a comment about where to find some)
- New feature: expandable code snippets that I think could come in handy here ✨ 
-  As mentioned in one of my comments, I think this guide would be better suited to be a code-oriented tutorial on architecture rather than a thought process/overview. I'll get back to you on the Web2 vs Web3 blogpost as this is quite similar where they both cover architecture

File: tutorials/eth-api/complete-dapp-architecture.md (hunk 2 above)
Comment (albertov19) [reply tutorials/eth-api/complete-dapp-architecture.md]: @eshaben are there parts of here we can include in the Web2 Web3 blogpost?

Comment (eshaben) [reply tutorials/eth-api/complete-dapp-architecture.md]: yeah we can always edit the web2 vs web3 blogpost

File: tutorials/eth-api/complete-dapp-architecture.md [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+This tutorial will use HardHat for managing smart contracts. You can initialize a project with HardHat using the following command:  
+
+```bash
+npx hardhat init
+```
+
+Before we start writing the smart contract, let's add a JSON-RPC URL to the config. Set the `hardhat.config.js` file with the following code, and replace `YOUR_PRIVATE_KEY` with your funded account's private key. This is for testing purposes, **never store your private key in plain text with real funds**.  
+
+```javascript
+require("@nomicfoundation/hardhat-toolbox");
```
[the remark this answers is not in the document]
Comment (eshaben) [reply tutorials/eth-api/complete-dapp-architecture.md]: I just had a tutorial where I initially used this, but Alberto pointed out that we don't need all the extra dependencies and features that come with Hardhat toolbox and we don't really use it any of our other tutorials. So I would just have the users install the hardhat ethers plugin and use that instead

File: tutorials/eth-api/complete-dapp-architecture.md (hunk 2 above)
Comment (albertov19) [reply tutorials/eth-api/complete-dapp-architecture.md]: Ok so @eshaben if you feel there is anything here that could add value to the blogpost I would go ahead and edit it - That way, Jeremy's effort were not in vain

**Review by eshaben — APPROVED** [review]
Sweeeet! Went through and made some grammar and formatting changes. Two major changes made were. The first being to the expandable snippets so they act as recaps of the files as code is added to them (to help avoid confusion). The second is that I also moved some of the content (i.e., setting up the ABI), so that the content is covered right before it is used (i.e., the ABI is used in the read smart contract data section)

Well done! This is a great addition to the docs site! 💜 🎉

**Pull request merged**